floss/language_identifier.py

```diff
@@ -0,0 +1,78 @@
+# Copyright (C) 2023 Mandiant, Inc. All Rights Reserved.
+import os
+
+import pefile
+
+import floss.logging_
+from floss.logging_ import DebugLevel
+
+logger = floss.logging_.getLogger("floss")
+
+
+def is_go_bin(sample: str) -> bool:
+    """
+    Check if the binary given is compiled with Go compiler or not
+    it checks the magic header of the pclntab structure -pcHeader-
+    the magic values varies through the version
+    reference:
+    https://github.com/0xjiayu/go_parser/blob/865359c297257e00165beb1683ef6a679edc2c7f/pclntbl.py#L46
+    """
+    try:
+        pe = pefile.PE(sample)
+    except pefile.PEFormatError as err:
+        logger.debug(f"NOT valid PE header: {err}")
+        return False
+    except IOError as err:
+        logger.error(f"File does not exist or cannot be accessed: {err}")
+        return False
+    except Exception as err:
+        logger.error(f"Unexpected error: {err}")
+        raise
+
+    go_magic = [
+        b"\xf0\xff\xff\xff\x00\x00",
+        b"\xfb\xff\xff\xff\x00\x00",
+        b"\xfa\xff\xff\xff\x00\x00",
+        b"\xf1\xff\xff\xff\x00\x00",
+    ]
+
+    # look for the .rdata section first
+    for section in pe.sections:
+        try:
+            section_name = section.Name.partition(b"\x00")[0].decode("utf-8")
+        except UnicodeDecodeError:
+            continue
+        if ".rdata" in section_name:
```

Comment:
I think you are right but two levels back? as if it is one level it will execute only one iteration and if the section not `.data` will return false. If 2 levels, it will return false if all the section checked. 
why `in` instead of `==`, I read a blog for a long time ago and he mentioned that the section name could be not strictly equal what we expect, it might have a garbage byte that will make the equality not true. It is better to use it anyway as the comparison will be small so will not affect the performance.

Reply:
But what about the loop? I guess it will return after only one iteration if it unindented one level.

Reply:
in which case, we only expect there to be a single section named .rdata, so we only expect the loop to even run once, so we can have the return in the inner branch, and in the outer scope just let things fall through. so maybe just remove the second return?

Reply:
```suggestion
        if ".rdata" == section_name:
```